### ISSUE TITLE
dist3 should be a blacklist, not a whitelist

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8761]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8761|
|Type|enhancement|
|Created|2016-08-16 01:45:19Z|
|Branch|https://github.com/twisted/twisted/tree/8761-dist3-blist|

We're at the point where we ought to make it a blacklist.

<details><summary>Searchable metadata</summary>

```
trac-id__8761 8761
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__8761_dist3_blist 8761-dist3-blist
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1471311919095961 1471311919095961
changetime__1473876174872954 1473876174872954
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

https://github.com/twisted/twisted/pull/501

Please review. Will need to compat policy except before merge, which I will put on the ML.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @hawkowl|
|-|-|

Thanks for fixing this, hawkowl.

This will make it easier for people to add new code without mistakenly forgetting to add it to dist3.

modulo compat policy and passing builds this looks mostly OK.  I put a few comments regarding the docstring on the github issue; please address those.  Love the diffstat: +230-770!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

Happy to see the blacklist :)

Can we have _dist3.py renamed to something like `_not_python3_ported.py` or something else? I find the `_dist3.py` very generic.

Also, I think that in this branch we should update the current documention / wiki in which twisted.python.dist3 is mentioned... here are some places https://github.com/twisted/twisted/pull/467/commits/0008cf1b123bd9e0a99293d622751486de676e43

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

and with the rename of dist3.py we should get in contact with the maintainer of the Twisted py3 port graph website. Not sure if the maintainer is on the ML.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl removed owner|
|-|-|

I updated the documentation, and detailed the policy that this blacklist implies.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph set owner to @hawkowl|
|-|-|

Doc changes look good.  Although all the cleanups could have been a different branch :).

Again, this is good to land as soon as we're through the compat process.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph commented|
|-|-|

adiroiban - since the py3 port website is an aid to twisted developers, it's fine to break compat with it ;).  The author is on the ML (although I believe they have other priorities at the moment).

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_9"></a>|@rodrigc commented|
|-|-|

I'm OK with this, merge it: https://github.com/twisted/twisted/pull/501#issuecomment-240904162

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl commented|
|-|-|

Will merge on thurs (when the compat exception clears)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset b920626cdd07a15f9c589c2d7bf4afaee494a485
```
#!CommitTicketReference repository="" revision="b920626cdd07a15f9c589c2d7bf4afaee494a485"
Merge 8761-dist3-blist: Make twisted.python.dist3 private API

Author: hawkowl
Reviewers: glyph, rodrigc
Fixes: #8761
```
